### ISSUE TITLE
Limit the amount of torrents queried

### ISSUE THREAD

**Issue by devos50** [Tribler/tribler#2779]
We should limit the amount of torrents queried by our torrent checker as indicated by the following report:

```
Traceback (most recent call last):
File "TriblerGUI/event_request_manager.py", line 91, in on_read_data
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "twisted/python/log.py", line 86, in callWithContext

File "twisted/python/context.py", line 118, in callWithContext

File "twisted/python/context.py", line 81, in callWithContext

File "twisted/internet/selectreactor.py", line 149, in _doReadOrWrite

--- <exception caught here> ---
File "twisted/internet/udp.py", line 249, in doRead

File "Tribler/Core/TorrentChecker/session.py", line 406, in datagramReceived

File "Tribler/Core/TorrentChecker/session.py", line 619, in handle_connection_response

File "Tribler/Core/TorrentChecker/session.py", line 389, in write_data

File "twisted/internet/udp.py", line 275, in write

twisted.internet.error.MessageLengthError: Message is too long to send: message too long.
```

**Comment by Dmole** [comment]
I would expect the message size limit can also be hit with just one torrent containing many files which is why APIs like [rtorrent](https://github.com/rakshasa/rtorrent/wiki/RPC-Setup-XMLRPC) by default only list IDs, instead of everything.
Optional fields can be included in the initial list and the list can be limited by fields like only list incomplete IDs. Per ID queries can obtain data to avoid one bug query... sorta SQL like in that columns are selectable and results can be limited by criteria. system.listMethods is also nice so people don't have to read the source code to figure out the API.

**Comment by synctext** [comment]
This issue crashed my Tribler every few minutes. Temporary hack is to remove your statedir / .Tribler directory.

**Comment by devos50** [comment]
Fixed by #2791
